### PR TITLE
Small refactor so we do less control flow

### PR DESCRIPTION
**Pull request by junpenglao**
Always append the state to the rightmost during progressive integration and only flip the 2 end when integration is terminated.

### CODE REVIEW

File: blackjax/inference/trajectory.py [hunk 1]
```diff
@@ -289,18 +280,29 @@ def add_one_state(expansion_state):
...
-            _,
             trajectory,
             termination_state,
             is_diverging,
             has_terminated,
         ) = jax.lax.while_loop(
             do_keep_integrating, add_one_state, initial_integration_state
         )
+        # In the while_loop we always extend on the right most direction.
+        new_trajectory = jax.lax.cond(
```
Comment (junpenglao) [inline blackjax/inference/trajectory.py:291]: @rlouf we can also fold this into `reorder_trajectories`, WDYT?

**Comment by rlouf** [comment]
Do you see any notable perf difference?

**Comment by junpenglao** [comment]
> Do you see any noticeable perf difference?

I feel the test runs faster but likely it is from your PR last round :-)

File: blackjax/inference/trajectory.py [hunk 2]
```diff
@@ -50,21 +50,15 @@ class Trajectory(NamedTuple):
     num_states: int
 
 
-def append_to_trajectory(
-    direction: int, trajectory: Trajectory, state: IntegratorState
+def rightmost_append_to_trajectory(
```
Comment (rlouf) [inline blackjax/inference/trajectory.py:53]: For every language I know of  `append`  = `add to the right` so we can probably name it `append_to_trajectory`.

File: blackjax/inference/trajectory.py [hunk 3]
```diff
@@ -50,21 +50,15 @@ class Trajectory(NamedTuple):
...
+    trajectory: Trajectory, state: IntegratorState
 ) -> Trajectory:
-    """Append a state to the trajectory to form a new trajectory."""
```
Comment (rlouf) [inline blackjax/inference/trajectory.py:56]: """Append a state to the (right of the) trajectory to form a new trajectory."""

File: blackjax/inference/trajectory.py (hunk 1 above)
Comment (rlouf) [reply blackjax/inference/trajectory.py:291]: I think that would make it uselessly complicated, we do not gain anything in terms of computation and it is a lot less explicit.

**Comment by rlouf** [comment]
Looks good to me. I only have two minor comments on the naming, ready to merge when it's fixed. Then I'll refactor the state passing and we can release.

**Comment by rlouf** [comment]
Great, merging.

**Pull request merged**